examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/readmes/GitHubProjectsReadme.md

```diff
@@ -0,0 +1,69 @@
+# GitHub Project Analysis with Batch Data
+
+In this sample, you'll see how to use [.NET for Apache Spark](https://dotnet.microsoft.com/apps/data/spark) 
+to analyze a file containing information about a set of GitHub projects. 
+
+## Problem
+
+Our goal here is to analyze information about GitHub projects. We start with some data prep (removing null data and unnecessary columns) 
+and then perform further analysis about the number of times different projects have been forked and how recently different projects
+have been updated.
+
+This sample is an example of **batch processing** since we're analyzing data that has already been stored and is not actively growing 
+or changing.
+
+## Solution
+
+Let's explore how we can use Spark to tackle this problem.
+
+### 1. Create a Spark Session
+
+In any Spark application, we need to establish a new SparkSession, which is the entry point to programming Spark with the Dataset and 
+DataFrame API.
+
+```CSharp
+SparkSession spark = SparkSession
+                .Builder()
+                .AppName("GitHub and Spark Batch")
+                .GetOrCreate();
+```
+
+By calling on the *spark* object created above, we can access Spark and DataFrame functionality throughout our program.
+
+### 2. Read Input File into a DataFrame
+
+Now that we have an entry point to the Spark API, let's read in our GitHub projects file. We'll store it in a DataFrame, while is a distributed collection of data organized into named columns.
+
+```CSharp
+DataFrame df = spark.Read().Text("Path to input data set");
+```
+
+### 3. Data Prep
+
+We can use `DataFrameNaFunctions` to drop rows with NA (null) values, and the `Drop()` method to remove certain columns from our data. 
+This helps prevent errors if we try to analyze null data or columns that do not matter for our final analysis.
+
+### 4. Further Analysis
+
+Spark SQL allows us to make SQL calls on our data. It is common to combine UDFs and Spark SQL so that we can apply a UDF to all 
+rows of our DataFrame.
+
+We can specifically call `spark.Sql` to mimic standard SQL calls seen in other types of apps, and we can also call methods like 
+`GroupBy` and `Agg` to specifically combine, filter, and perform calculations on our data.
+
+### 5. Running Your App
+
+Checkout the directions for building and running this app on [Windows](../../../../../docs/building/windows-instructions.md) or [Ubuntu](../../../../../docs/building/ubuntu-instructions.md).
+
+#### Windows Example:
+
+```CSharp
+spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner --master local C:\GitHub\spark\src\scala\microsoft-spark-2.4.x\target\microsoft-spark-2.4.x-0.6.0.jar Microsoft.Spark.CSharp.Examples.exe Sql.Batch.GitHubProjects C:\Users\MyUserName\Documents\projects_smaller.csv
```

Comment:
Is there a better approach than having version (`0.6.0`), so that we don't have to update the doc every time for each release? Maybe just `<path/to/microsoft-spark-<version>.jar>` or something similar? Since we have a link to building/running this app, it may be OK.

Reply:
Perhaps I could add a note explaining to update `0.6.0` to the latest/desired version of Spark.NET? 

I wanted to keep the sample code here as accurate to an actual submission as possible since it makes it a lot easier for users to see an actual code example rather than having to think more about "What jar file should I be finding? What are valid version numbers? What separate doc do I have to go visit to figure this out?"

But if you think it'd be a lot better to avoid using actual jar path here, I'm open to changing it. That's just my POV as a user who did have to take some time to figure this out. 😄

Reply:
I think the only issue with having version numbers within documentation is that the versions can get outdated quickly and there can be a lot of places that need to be updated every version update.

Reply:
Users will have to point to a local path on their file system regardless, so they will still have to massage the command a bit.     I would do something like `/path/to/microsoft-spark-<version>.jar` and `/path/to/projects_smaller.csv`

Reply:
Makes sense. Updated the path to the jar file and projects_smaller.